### PR TITLE
use "std::regex_replace" instead of obsolete pcre

### LINKED ISSUE #2: Migrate from pcre to pcre2
Request that pcre is migrated to pcre2.
With pcre being (nearing) obsolescence - could xmltv2vdr follow suit and update to pcre2.
It does look like the wrapper at https://github.com/jpcre2/jpcre2 might be a candidate dropping for pcrecpp?

### PR DESCRIPTION
**Pull request by heitbaum**
Update to using std::regex as PCRE is obsolete and being removed from distributions. 
Needs to be run tested by a vdr-plugin-xmltv2vdr regex user.

- ref: https://github.com/PCRE2Project/pcre2/issues/51
- Closes #2

### CODE REVIEW

**Comment by heitbaum** [comment]
@M-Reimer could you please take a look - and if ok merge

**Comment by M-Reimer** [comment]
I can't approve this as you explicitly write that this `Needs to be run tested by a vdr-plugin-xmltv2vdr regex user.` which I am not. I will not even be able to set up any proper test environment for this.

Forwarded to here: https://www.vdr-portal.de/forum/index.php?thread/135994-nutzer-von-xmltv2vdr-gesucht-die-einen-pr-testen-k%C3%B6nnen/

If noone wants to test this, I would rather leave this open. At least on Arch there are still 66 packages which depend on pcre:
https://archlinux.org/packages/core/x86_64/pcre/
If Arch finally manages to get rid of pcre and still noone wanted to test your change, I would merge without testing.

File: dist/epgdata2xmltv/epgdata2xmltv.cpp
Comment (M-Reimer) [inline dist/epgdata2xmltv/epgdata2xmltv.cpp:1]: I guess I could just accept this change as it is no longer possible to test it! This keeps this example program somewhat usable.
This grabber (epgdata2xmltv) was used to grab EPG data from a subscription based service (epgdata.com) which is no longer in service https://www.epgdata.com/
If you are actually a maintainer of a distribution package for xmltv2vdr, then you should stop building and shipping epgdata2xmltv. I will leave the code in as an example but probably add a README.md that clearly states that the end user has no benefit if this program is still shipped. The user has to provide his own grabber, probably written in a language better suited for such tasks (Python, ...)

File: event.cpp
Comment (M-Reimer) [inline event.cpp:537]: I have no clue what this code actually should do as I never did RegEx in C(++) but let's give it a try and maybe you can correct me:

1. replaces every occurence of a single quote (') with two single quotes ('') (doubles up the "'" character).
2. replaces "^" with a single quote ('). We have to keep in mind that the escaping, done here, is handled by the RegEx library as (^) is a control character in RegEx, if not excaped. So it actually just replaces "^" and we can ignore the "\\".
3. replaces ('NULL') with (NULL) (removes single quotes).

Maybe I'm totally off here, but couldn't this be all done with a simple string replacement? Why RegEx?

BUT there probably is an issue with your "was something changed" check. The first replacement increases length, the second and third one decreases length. So it would be possible to get into a case where this cancels out and some replacements were made but not caught by the "if" statement.

Maybe the only reason, why RegEx was used here, is to know if changes were made. Going over length is a possibility but only if the potential issue above is fixed.

File: import.cpp
Comment (M-Reimer) [inline import.cpp:1409]: Similar case here. Replaces all (') with ('') (doubles them up). Here the length based check will work as it does not cover several replaces. But if length is used to verify if changes were made, then a "regular string replace" should do.

File: event.cpp
Comment (M-Reimer) [inline event.cpp:548]: See above

File: import.cpp
Comment (M-Reimer) [inline import.cpp:1519]: See above

File: import.cpp
Comment (M-Reimer) [inline import.cpp:1657]: See above

**Review by M-Reimer — CHANGES_REQUESTED** [review]
I did an attempt in reviewing this.

Probably with a bit of rework the regex usage can go in the plugin and only is needed in the (example, only for demonstration) epgdata2xmltv application.

**Comment by M-Reimer** [comment]
Got a improved PR from the initial developer which fixes a few issues with your PR.

**Comment by heitbaum** [comment]
> Got a improved PR from the initial developer which fixes a few issues with your PR.

Thank you both @M-Reimer @joed74

- I have updated https://github.com/LibreELEC/LibreELEC.tv/pull/8490 against 089976d4950c2641714be599766f0534825165ca

**Pull request closed without merging**
